src/MultiFormats.Tests/MultiHashTest.cs

```diff
@@ -0,0 +1,535 @@
+            Assert.IsTrue(mh2.Matches(hello));
+            Assert.IsFalse(mh2.Matches(hello1));
+
+            var mh3 = MultiHash.ComputeHash(hello, "keccak-512");
+            Assert.IsTrue(mh3.Matches(hello));
+            Assert.IsFalse(mh3.Matches(hello1));
+        }
+
+        [TestMethod]
+        public void Matches_Stream()
+        {
+            var hello = new MemoryStream(Encoding.UTF8.GetBytes("Hello, world."));
+            var hello1 = new MemoryStream(Encoding.UTF8.GetBytes("Hello, world"));
+            hello.Position = 0;
+            var mh = MultiHash.ComputeHash(hello);
+
+            hello.Position = 0;
+            Assert.IsTrue(mh.Matches(hello));
+
+            hello1.Position = 0;
+            Assert.IsFalse(mh.Matches(hello1));
+        }
+
+        [TestMethod]
+        public void Wire_Formats()
+        {
+            var hashes = new[]
+            {
+                "5drNu81uhrFLRiS4bxWgAkpydaLUPW",                                                             // sha1
+                "QmaozNR7DZHQK1ZcU9p7QdrshMvXqWK6gpu5rmrkPdT3L4",                                             // sha2_256
+                "8Vtkv2tdQ43bNGdWN9vNx9GVS9wrbXHk4ZW8kmucPmaYJwwedXir52kti9wJhcik4HehyqgLrQ1hBuirviLhxgRBNv", // sha2_512
+            };
+            var helloWorld = Encoding.UTF8.GetBytes("hello world");
+            foreach (var hash in hashes)
+            {
+                var mh = new MultiHash(hash);
+                Assert.IsTrue(mh.Matches(helloWorld), hash);
+            }
+        }
+
+        [TestMethod]
+        public void To_String_Is_Base58_Representation()
+        {
+            var hash = "QmaozNR7DZHQK1ZcU9p7QdrshMvXqWK6gpu5rmrkPdT3L4";
+            var mh = new MultiHash(hash);
+            Assert.AreEqual(hash, mh.ToString());
+        }
+
+        [TestMethod]
+        public void Implicit_Conversion_From_String()
+        {
+            var hash = "QmaozNR7DZHQK1ZcU9p7QdrshMvXqWK6gpu5rmrkPdT3L4";
+            MultiHash mh = hash;
+            Assert.IsNotNull(mh);
+            Assert.IsInstanceOfType(mh, typeof(MultiHash));
+            Assert.AreEqual(hash, mh.ToString());
+        }
+
+        [TestMethod]
+        public void Value_Equality()
+        {
+            var a0 = new MultiHash("QmaozNR7DZHQK1ZcU9p7QdrshMvXqWK6gpu5rmrkPdT3L4");
+            var a1 = new MultiHash("QmaozNR7DZHQK1ZcU9p7QdrshMvXqWK6gpu5rmrkPdT3L4");
+            var b = new MultiHash("QmaozNR7DZHQK1ZcU9p7QdrshMvXqWK6gpu5rmrkPdT3L5");
+            MultiHash c = null;
+            MultiHash d = null;
+
+            Assert.IsTrue(c == d);
+            Assert.IsFalse(c == b);
+            Assert.IsFalse(b == c);
+
+            Assert.IsFalse(c != d);
+            Assert.IsTrue(c != b);
+            Assert.IsTrue(b != c);
+
+#pragma warning disable 1718
+            Assert.IsTrue(a0 == a0);
+            Assert.IsTrue(a0 == a1);
+            Assert.IsFalse(a0 == b);
+
+#pragma warning disable 1718
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Do not suppress issues.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)